### ISSUE TITLE
Wrong thread-safety trait implementing on LLVMString

### ISSUE THREAD

**Issue by GCPanic** [TheDan64/inkwell#488]
**Describe the Bug**
Currently, [inkwell](https://thedan64.github.io/inkwell/inkwell/index.html)::[support](https://thedan64.github.io/inkwell/inkwell/support/index.html)::[LLVMString](https://thedan64.github.io/inkwell/inkwell/support/struct.LLVMString.html#) implementing `!Send`, `!Sync` and `Deref<Target = CStr>`. This is clearly incorrect, as `CStr` implementing `Send` and `Sync`.

**To Reproduce**
Unrelated

**Expected Behavior**
`LLVMString` is just an `CStr` with custom `Drop`, so it should be safe to send it's reference to other threads.

On the other hands, i've searched some informations about LLVM Messages, and i don't found any documents saying it's not thread-safe. In this case, `LLVMString` should implement `Send` as well.

**LLVM Version (please complete the following information):**
 - LLVM Version: [Unrelated]
 - Inkwell Branch Used: [Unrelated]

**Desktop (please complete the following information):**
 - OS: [Unrelated]

**Additional Context**
None

**Comment by TheDan64** [comment]
I agree that I think `LLVMString` should be `Send`, but I think your analysis is slightly off in that it's more like `CString`, not `CStr`. `CStr` (which is like `&str`) shouldn't be `Send` nor `Sync`.

**Comment by GCPanic** [comment]
> it's more like `CString`, not `CStr`. 

Yes, you're right. LLVMString has string's ownership, it more like a CString.

> &CStr (which is like &str) shouldn't be Send nor Sync when the reference is non static.

Not exactly, although thread::spawn does ask for values to be 'static, Send and Sync trait has nothing to do with lifetimes. And, infact, CStr does implements Send and Sync(which means &CStr also implements them)

**Comment by TheDan64** [comment]
Ah yeah, good point
